kotlin-checks-test-sources/src/main/kotlin/checks/InterfaceCouldBeFunctionalCheckSample.kt

```diff
@@ -0,0 +1,125 @@
+package checks
+
+class InterfaceCouldBeFunctionalCheckSample {
+
+    interface IntMapper<T> { // Noncompliant {{Interface should be functional or replaced with a functional type.}}
+        fun map(value: Int): T
+    }
+
+    fun interface FunctionalIntMapper<T> { // Compliant
+        fun map(value: Int): T
+    }
+
+    @FunctionalInterface // Noncompliant {{"@FunctionalInterface" annotation has no effect in Kotlin}}
+    interface IntMapperWithWrongAnnotation<T> { // Noncompliant {{Interface should be functional or replaced with a functional type.}}
+        fun map(value: Int): T
+    }
+
+    @FunctionalInterface // Noncompliant {{"@FunctionalInterface" annotation has no effect in Kotlin}}
+    fun interface FunctionalIntMapperWithWrongAnnotation<T> { // Compliant
+        fun map(value: Int): T
+    }
+
+    @AnyOtherAnnotation // Compliant
+    interface WithAnyOtherAnnotation<T> { // Noncompliant {{Interface should be functional or replaced with a functional type.}}
+        fun map(value: Int): T
+    }
+
+    @AnyOtherAnnotation // Compliant
+    fun interface FunctionalInterfaceAnyOtherAnnotation<T> { // Compliant
+        fun map(value: Int): T
+    }
+
+    interface EmptyInterface { // Compliant
+    }
+
+    annotation class AnyOtherAnnotation
+
+    interface TwoFunctionInterface<T> { // Compliant
+        fun map(value: Int): T
+        fun nap(value: Int): T
+    }
+
+    interface FunctionAndPropertyInterface<T> { // Compliant
+        fun map(value: Int): T
+        val limit: Int
+    }
+
+    abstract class AbstractClass<T> { // Compliant
+        abstract fun map(value: Int): T
+    }
+
+    class NonAbstractClass { // Compliant
+        fun map(value: Int) = value * value
+    }
+
+    interface NonFunctionalOuterInterface { // Noncompliant {{Interface should be functional or replaced with a functional type.}}
+
+        fun map(value: Int): Int
+
+        interface NonFunctionalInnerInterface { // Noncompliant {{Interface should be functional or replaced with a functional type.}}
```

Comment:
It's nice to check issue locations at least once per different issue (1 for interface keyword and 1 for @FunctionalInterface annotation)